subsys/shell/shell_ops.c

```diff
@@ -485,6 +485,11 @@ void z_shell_vt100_colors_restore(const struct shell *sh,
 	vt100_bgcolor_set(sh, color->bgcol);
 }
 
+static int z_shell_get_return_value(const struct shell *sh)
```

Comment:
This is not right and has not even been tested, 
"warning: implicit declaration of function 'z_shell_get_return_value';"
"warning: 'z_shell_get_return_value' defined but not used",
it builds because of "SHELL_COND_CMD_ARG_REGISTER(SHELL_CMDS_RETURN_VALUE,", where `SHELL_CMDS_RETURN_VALUE` should be `CONFIG_SHELL_CMDS_RETURN_VALUE`.

Necessary changes:
```diff
diff --git a/subsys/shell/shell_ops.c b/subsys/shell/shell_ops.c
index 4ae957eed3c..a60f9de7acd 100644
--- a/subsys/shell/shell_ops.c
+++ b/subsys/shell/shell_ops.c
@@ -485,7 +485,7 @@ void z_shell_vt100_colors_restore(const struct shell *sh,
        vt100_bgcolor_set(sh, color->bgcol);
 }
 
-static int z_shell_get_return_value(const struct shell *sh)
+int z_shell_get_return_value(const struct shell *sh)
 {
        return sh->ctx->ret_val;
 }
diff --git a/subsys/shell/shell_ops.h b/subsys/shell/shell_ops.h
index 39b66014360..d0e7fef6929 100644
--- a/subsys/shell/shell_ops.h
+++ b/subsys/shell/shell_ops.h
@@ -345,6 +345,8 @@ static inline void z_shell_vt100_colors_store(const struct shell *sh,
 void z_shell_vt100_colors_restore(const struct shell *sh,
                                  const struct shell_vt100_colors *color);
 
+int z_shell_get_return_value(const struct shell *sh);
+
 /* This function can be called only within shell thread but not from command
  * handlers.
  */
```

Reply:
You are correct, this was not tested after my last git rebase/history rewrite. I moved the function to be a static inline function in the header. It now works correctly. Build tested for mpfs_icicle.